tests/testthat/test-layer_predict.R

```diff
@@ -31,3 +31,47 @@ test_that("prediction with interval works", {
   expect_equal(nrow(p), 108L)
   expect_named(p, c("geo_value", "time_value", ".pred_lower", ".pred_upper"))
 })
+
+test_that("layer_predict dots validation", {
+  # We balk at unnamed arguments, though perhaps not with the most helpful error messages:
+  expect_error(
+    frosting() %>% layer_predict("pred_int", list(), tibble::tibble(x = 5)),
+    class = "epipredict__layer_predict__unnamed_dot"
+  )
+  expect_error(
+    frosting() %>% layer_predict("pred_int", list(), "maybe_meant_to_be_id"),
+    class = "epipredict__layer_predict__unnamed_dot"
+  )
+  # We allow arguments that might actually work at prediction time:
+  expect_no_error(frosting() %>% layer_predict(type = "quantile", interval = "confidence"))
+
+  # We don't detect completely-bogus arg names until predict time:
+  expect_no_error(f_bad_arg <- frosting() %>% layer_predict(bogus_argument = "something"))
+  wf_bad_arg <- wf %>% add_frosting(f_bad_arg)
+  expect_error(predict(wf_bad_arg, latest))
+  # Some argument names only apply for some prediction `type`s; we don't check for ignored arguments, and neither does workflows:
+  expect_no_error(frosting() %>% layer_predict(eval_time = "preferably this would error"))
+
+  # ^ (currently with a truly awful error message, due to an extra comma in parsnip::check_pred_type_dots)
+  #
+  # Unfortunately, we outright ignore attempts to pass args via `predict.epi_workflow`:
+  f_predict <- frosting() %>% layer_predict()
+  wf_predict <- wf %>% add_frosting(f_predict)
```

Comment:
Do we want `...` from `predict.epi_workflow()` to be forwarded only to `slather.layer_predict()` or to all the other layer `slather` methods?  Would other layers ever need to know about these args?

(If we do end up forwarding to all `slather.layer_*`, we should make sure the only args allowed in `slather.layer_*` are documented (correctly) in `predict.epi_workflow()` to prevent name collisions from inducing crazy behavior.)

.... Just from thinking about this, I think there's probably a bunch of errors in the current implementation, and there's going to be a bunch of extra logic that might be required.  E.g., for `layer_population_scaling` to work correctly, it needs to know what the prediction output columns are.  But that changes based on the prediction type!  Do we have roles of columns tracked in the prediction output that might allow this to be automated?  I assume not, right?

Reply:
Do we want to continue down this sort of path & fix up issues, or instead disallow `type` etc. and make them use layers instead?

Reply:
If we forward, I think only to `layer_predict()`. And perhaps only in the case that that's the only layer. Those arguments could potentially result in arbitrary structures in the predictions data frame, so I think it would be near impossible to anticipate them all. Perhaps additional layers could be written specifically to handle some of these options in the future.

Reply:
Okay, makes sense.  For this PR, I'll probably hack on the "only in the case that that's the only layer" part in a minimal fashion.

brainstorm: if we add processing to other layers for these, we probably want a bit of refactoring:
- Add an extra required argument to `slather()` to hold the dots list, instead of passing via dots.
- Pass dots list to every layer.
- Validate layer support by either:
  - Checking for an empty dots list in each unsupported `slather()` --- initially, we'd add such a check at the beginning of every `slather.layer_*()` other than `slather.layer_predict()`.
  - Have an S3 generic (e.g., "`predict_dots_supported(layer)`") that outputs whether/which predict dots arguments are supported, and use that in `predict.epi_workflow()`.  This is more complex but could list all the unsupported layers at once and may help us/devs not forget to check for emptiness.
  - Have a fixed list of layers with support that we check against in `predict.epi_workflow()`.  Not extensible.

Reply:
Should this settings-forwarding also be supported by `forecast`?  Or do we want to push people toward more-completely-supported features there?

Reply:
Let's say "completely supported features for now"

Reply:
todo: looks like code comment above shouldn't apply now; revise and check if test is also wrong now